lib/smart_answer_flows/vat-payment-deadlines.rb

```diff
@@ -9,15 +9,14 @@ def define
 
       date_question :when_does_your_vat_accounting_period_end? do
         default_day { -1 }
-        next_node do
-          question :how_do_you_want_to_pay?
-        end
 
-        calculate :period_end_date do |response|
-          date = response
-          raise InvalidResponse unless date == date.end_of_month
+        on_response do |response|
+          self.calculator = Calculators::VatPaymentDeadlines.new
+          calculator.period_end_date = response
+        end
```

Comment:
Think we should be using a validate block to make it clear when we do user input validations - https://github.com/alphagov/smart-answers/blob/master/doc/smart-answers/flow-definition.md#validatemessage_key-block

Reply:
I'll change it to a validate block